src/aiida/engine/daemon/execmanager.py

```diff
@@ -20,11 +20,10 @@
 from logging import LoggerAdapter
 from pathlib import Path
 from tempfile import NamedTemporaryFile, TemporaryDirectory
-from typing import TYPE_CHECKING, Any, List, Optional, Tuple, Union
+from typing import TYPE_CHECKING, Any, List, Optional, Tuple, Union, assert_never
```

Comment:
`assert_never` is only available since Python 3.11, we'd need a fallback import from `typing_extensions` for python 3.9 and 3.10.

https://typing.python.org/en/latest/guides/unreachable.html#assert-never-and-exhaustiveness-checking

Reply:
I've pushed a commit to implement this